### PR TITLE
Use f-strings

### LINKED ISSUE #4632: Replace old style string interpolation with f strings
Very minor issue, change `%` style interpolation to f string format. Although equally functional it's inconsistent with the fstring interpolation used elsewhere in the file and for someone newer to python could be a small tripping point

https://github.com/pymc-devs/pymc3/blob/v4/pymc3/model.py#L1125

### PR DESCRIPTION
**Pull request by adityaraute**
Fixes #4632 

**Thank your for opening a PR!**

Depending on what your PR does, here are a few things you might want to address in the description:
+ [ ] what are the (breaking) changes that this PR makes?
+ [x] important background, or details about the implementation
Simple change with interpolation.
+ [ ] are the changes—especially new features—covered by tests and docstrings?
+ [ ] [linting/style checks have been run](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Python-Code-Style)
+ [ ] [consider adding/updating relevant example notebooks](https://github.com/pymc-devs/pymc-examples)
+ [ ] right before it's ready to merge, mention the PR in the RELEASE-NOTES.md

### CODE REVIEW

File: pymc3/model.py [hunk 1]
```diff
@@ -1122,7 +1122,11 @@ def add_random_variable(self, var, dims=None):
 
     @property
     def prefix(self):
-        return "%s_" % self.name if self.name else ""
+        if self.name:
+            val = self.name
+            return f"{val}_"
+        else: 
+            return ""
```
Comment (OriolAbril) [inline pymc3/model.py]:
```suggestion
        return f"{self.name}_" if self.name else ""
```

**Review by OriolAbril — CHANGES_REQUESTED** [review]
I believe the issue uses this line as an example of the kind of sting formatting operations that need to be updated, not that this is the _only_ like where old formatting is used. @canyon289 ?

(the changes requested are about the modified code, not about this scope question)

File: pymc3/model.py (hunk 1 above)
Comment (adityaraute) [reply pymc3/model.py]: Oh is it? OK maybe I didnt understand it well enough

**Review by OriolAbril — COMMENTED** [review]
The fix is fine, but after a quick search, there are still many places where old formatting is used. If you want to close the issue, all occurrences of the old formatting should be updated, if you want to partially address the issue (and therefore keep it open for someone else to continue working on this) then it's already fine.

Note that you can find old formatting quickly with `grep % pymc3/ -r` for example. There will be some false positives of actual percentages or modulo operator, but you can see in the small portion of the result copied below that there are still many places where old formatting is used.

```
....
pymc3/backends/report.py:                elif start <= warn.step < stop and (warn.step - start) % step == 0:
pymc3/backends/base.py:                    raise ValueError("Sampler statistic %s appears with " "different types." % key)
pymc3/backends/base.py:            raise KeyError("Unknown sampler stat %s" % stat_name)
pymc3/backends/base.py:        raise KeyError("Unknown variable %s" % var)
pymc3/backends/base.py:            raise KeyError("Unknown sampler statistic %s" % stat_name)
pymc3/backends/ndarray.py:        raise TraceDirectoryError("%s is not a PyMC3 saved chain directory." % directory)
pymc3/backends/ndarray.py:            raise TraceDirectoryError("%s is not a trace directory" % self.directory)
pymc3/backends/arviz.py:                    "posterior predictive variable %s's shape not compatible with number of chains and draws. "
pymc3/data.py:            raise TypeError("Unrecognized size type, %r" % size)
pymc3/data.py:                    raise TypeError("Unrecognized size type, %r" % user_size)
pymc3/data.py:            raise TypeError("Unrecognized size type, %r" % user_size)
pymc3/data.py:                    "size and random seed are both ints, got %r" % batch_size
pymc3/data.py:                        "Double Ellipsis in `batch_size` is restricted, got %r" % batch_size
pymc3/data.py:                    "number of ints is bigger that ndim, got %r" % batch_size
pymc3/data.py:            raise TypeError("Unrecognized size type, %r" % batch_size)

```

**Comment by adityaraute** [comment]
> The fix is fine, but after a quick search, there are still many places where old formatting is used. If you want to close the issue, all occurrences of the old formatting should be updated, if you want to partially address the issue (and therefore keep it open for someone else to continue working on this) then it's already fine.
> 
I'll try to work on it and will update soon.

**Comment by canyon289** [comment]
Thanks @adityaraute and @OriolAbril. @adityaraute know that we'll hold this PR for merge until we finish our v4 version so if you want to take it slow on that would be fine!

**Comment by michaelosthege** [comment]
@canyon289 if this PR goes into `v4`, and as long as there are no conflicts I don't think we need to hold it.
Only PRs into `master` are on hold until we complete https://github.com/pymc-devs/pymc3/milestone/24

**Comment by adityaraute** [comment]
Close this PR if #4671  is merged.

**Comment by michaelosthege** [comment]
#4671 was merged

**Pull request closed without merging**